cmd/copy.go

```diff
@@ -490,21 +490,21 @@ func (raw rawCopyCmdArgs) cookWithId(jobId common.JobID) (cookedCopyCmdArgs, err
 		}
 	case common.EFromTo.BlobBlob(),
 		common.EFromTo.FileBlob(),
+		common.EFromTo.FileFile(),
+		common.EFromTo.BlobFile(),
 		common.EFromTo.S3Blob():
 		if cooked.preserveLastModifiedTime {
 			return cooked, fmt.Errorf("preserve-last-modified-time is not supported while copying from service to service")
 		}
 		if cooked.followSymlinks {
 			return cooked, fmt.Errorf("follow-symlinks flag is not supported while copying from service to service")
 		}
+		// blob type is not supported if we are not doing blob -> blob
+		if cooked.blobType != common.EBlobType.Detect() && cooked.fromTo != common.EFromTo.BlobBlob() {
+			return cooked, fmt.Errorf("blob-type is not supported for the scenario (%s)", cooked.fromTo.String())
+		}
```

Comment:
This seems odd. I thought we supported the following:
* Blob - > Blob, all blobType parameter values are valid, and Detect means "Detect the type of the source blob"
* anything else -> Blob, all blobType parameter values are valid, and Detect means "Look for VHD/VHDX extension"
* anything -> NOT Blob, _only_ Detect is valid, and it means "No blob type specified"

I think the code here is incorrectly treating the last two cases, above, as being the same.

Reply:
Oh, I see that you've based this code on some code removed from a few lines below.  I believe that code was incorrect. We actually had a bug logged about it recently (via internal email).

Reply:
I see, I don't recall seeing the bug report, but what you are saying makes sense to me.

The existing code gave me the impression that S3/File -> Blob can only be Block Blobs.